pkg/virt-operator/install-strategy/create.go

```diff
@@ -0,0 +1,995 @@
+			}
+			log.Log.V(2).Infof("clusterrolebinding %v created", crb.GetName())
+		} else if !objectMatchesVersion(&cachedCrb.ObjectMeta, imageTag, imageRegistry) {
+			// Update existing, we don't need to patch for rbac rules.
+			_, err = rbac.ClusterRoleBindings().Update(crb)
+			if err != nil {
+				return false, fmt.Errorf("unable to update clusterrolebinding %+v: %v", crb, err)
+			}
+			log.Log.V(2).Infof("clusterrolebinding %v updated", crb.GetName())
+
+		} else {
+			log.Log.V(4).Infof("clusterrolebinding %v already exists", crb.GetName())
+		}
+	}
+
+	// create/update Roles
+	for _, r := range targetStrategy.roles {
+		var cachedR *rbacv1.Role
+
+		r := r.DeepCopy()
+		obj, exists, _ := stores.RoleCache.Get(r)
+		if exists {
+			cachedR = obj.(*rbacv1.Role)
+		}
+
+		injectOperatorLabelAndAnnotations(&r.ObjectMeta, imageTag, imageRegistry)
+		if !exists {
+			// Create non existent
+			expectations.Role.RaiseExpectations(kvkey, 1, 0)
+			_, err := rbac.Roles(kv.Namespace).Create(r)
+			if err != nil {
+				expectations.Role.LowerExpectations(kvkey, 1, 0)
+				return false, fmt.Errorf("unable to create role %+v: %v", r, err)
+			}
+			log.Log.V(2).Infof("role %v created", r.GetName())
+		} else if !objectMatchesVersion(&cachedR.ObjectMeta, imageTag, imageRegistry) {
+			// Update existing, we don't need to patch for rbac rules.
+			_, err = rbac.Roles(kv.Namespace).Update(r)
+			if err != nil {
+				return false, fmt.Errorf("unable to update role %+v: %v", r, err)
+			}
+			log.Log.V(2).Infof("role %v updated", r.GetName())
+
+		} else {
+			log.Log.V(4).Infof("role %v already exists", r.GetName())
+		}
+	}
+
+	// create/update RoleBindings
+	for _, rb := range targetStrategy.roleBindings {
+		var cachedRb *rbacv1.RoleBinding
+
+		rb := rb.DeepCopy()
+		obj, exists, _ := stores.RoleBindingCache.Get(rb)
+
+		if exists {
+			cachedRb = obj.(*rbacv1.RoleBinding)
+		}
+
+		injectOperatorLabelAndAnnotations(&rb.ObjectMeta, imageTag, imageRegistry)
+		if !exists {
+			// Create non existent
+			expectations.RoleBinding.RaiseExpectations(kvkey, 1, 0)
+			_, err := rbac.RoleBindings(kv.Namespace).Create(rb)
+			if err != nil {
+				expectations.RoleBinding.LowerExpectations(kvkey, 1, 0)
+				return false, fmt.Errorf("unable to create rolebinding %+v: %v", rb, err)
+			}
+			log.Log.V(2).Infof("rolebinding %v created", rb.GetName())
+		} else if !objectMatchesVersion(&cachedRb.ObjectMeta, imageTag, imageRegistry) {
+			// Update existing, we don't need to patch for rbac rules.
+			_, err = rbac.RoleBindings(kv.Namespace).Update(rb)
+			if err != nil {
+				return false, fmt.Errorf("unable to update rolebinding %+v: %v", rb, err)
+			}
+			log.Log.V(2).Infof("rolebinding %v updated", rb.GetName())
+
+		} else {
+			log.Log.V(4).Infof("rolebinding %v already exists", rb.GetName())
+		}
+	}
+
+	// create/update Services
+	for _, service := range targetStrategy.services {
```

Comment:
Service Updates will not work as expected in case we remove a port for update or rollback.